src/java/com/sarality/app/view/dialog/DialogActionPerformer.java

```diff
@@ -0,0 +1,25 @@
+package com.sarality.app.view.dialog;
+
+import android.content.DialogInterface;
+
+import com.sarality.app.view.action.Action;
+
+/**
+ * Utility class to perform the action.
+ * <p/>
+ * This is the class that gets registered with the dialog and delegates the call to the actual action.
+ *
+ * @author abhideep@ (Abhideep Singh)
+ */
+class DialogActionPerformer implements DialogInterface.OnCancelListener {
+  private final Action<DialogActionContext> action;
+
+  DialogActionPerformer(Action<DialogActionContext> action) {
+    this.action = action;
+  }
+
+  @Override
+  public void onCancel(DialogInterface dialog) {
+    action.performAction(new DialogActionContext(dialog));
```

Comment:
Yup on Dimiss was added much later. We can add that support if we start seeing the need for it, but will have to add an additional support libraray for that v7 or v11 I think